src/lib.rs

```diff
@@ -293,6 +300,18 @@ impl<'a> Iterator for Iter<'a> {
     }
 }
 
+/// Iterator over the string idtenfiers of the protocols (not addrs) in a multiaddr
+pub struct ProtoStackIter<'a> {
+    parts: Iter<'a>,
+}
+
+impl<'a> Iterator for ProtoStackIter<'a> {
+    type Item = &'static str;
+    fn next(&mut self) -> Option<Self::Item> {
+        self.parts.next().map(|p| p.tag())
```

Comment:
```suggestion
        self.parts.next().map(Protocol::tag)
```

This should work, right?

Reply:
`tag()` takes `&self` rather than `self`, so to make that work I needed as_ref, but even so I agree this approach is better.